src/diamond/handler/signalfx_handler.py

```diff
@@ -5,44 +5,50 @@
 
 #### Dependencies
 
- * urllib2
+ * signalfx (pip install signalfx)
 
 
 #### Configuration
 Enable this handler
 
- * handers = diamond.handler.httpHandler.SignalfxHandler
+ * handlers = diamond.handler.signalfx_handler.SignalfxHandler
 
  * auth_token = SIGNALFX_AUTH_TOKEN
+
  * batch_size = [optional | 300 ] will wait for this many requests before
      posting
 """
-
 from Handler import Handler
-from diamond.util import get_diamond_version
-import json
+import collections
 import logging
 import time
-import urllib2
+
+try:
+    import signalfx
+except ImportError:
+    raise ImportError("Failed to load signalfx module. "
+                      "Install signalfx module - `pip install signalfx`")
+
+from diamond.util import get_diamond_version
 
 
 class SignalfxHandler(Handler):
 
     # Inititalize Handler with url and batch size
     def __init__(self, config=None):
-        Handler.__init__(self, config)
+        super(SignalfxHandler, self).__init__(config)
         self.metrics = []
+        auth_token = self.config['auth_token']
+        url = self.config['url']
         self.batch_size = int(self.config['batch'])
-        self.url = self.config['url']
-        self.auth_token = self.config['auth_token']
-        self.batch_max_interval = self.config['batch_max_interval']
+        self.batch_max_interval = float(self.config['batch_max_interval'])
         self.resetBatchTimeout()
-        if self.auth_token == "":
-            logging.error("Failed to load Signalfx module")
-            return
+        self.signalfx = signalfx.SignalFx(auth_token, ingest_endpoint=url,
```

Comment:
Looks like this uses the default 1s timeout. Making api post timeout a configuration option for the handler would be very useful.